servers/src/main/java/tachyon/worker/block/evictor/LRUEvictor.java

```diff
@@ -0,0 +1,140 @@
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+
+import tachyon.Constants;
+import tachyon.Pair;
+import tachyon.worker.BlockStoreLocation;
+import tachyon.worker.block.BlockAccessEventListener;
+import tachyon.worker.block.BlockMetadataManager;
+import tachyon.worker.block.meta.BlockMeta;
+
+public class LRUEvictor implements Evictor, BlockAccessEventListener {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private final BlockMetadataManager mMeta;
+
+  /** Double-Link List, most recently accessed block is at tail of the list */
+  private class Node extends DoubleLinkListNode {
+    long blockId;
+
+    Node() {
+      blockId = -1;
+      prev = next = null;
+    }
+
+    Node(long blkId) {
+      blockId = blkId;
+      prev = next = null;
+    }
+
+    Node nextNode() {
+      return (Node) next;
+    }
+  }
+
+  private Node mHead;
+  private Node mTail;
+  /** Map from blockId to corresponding Node in the Double-Link List */
+  private Map<Long, Node> mCache;
+
+
+  public LRUEvictor(BlockMetadataManager meta) {
+    mMeta = meta;
+
+    mHead = new Node();
+    mTail = new Node();
+    mHead.append(mTail);
+    mCache = new HashMap<Long, Node>();
+  }
+
+  /**
+   * Free space in the given block store location. The location can be a specific location, or
+   * {@link BlockStoreLocation#anyTier()} or {@link BlockStoreLocation#anyDirInTier(int)} .
+   *
+   * If the total free space is fewer than {@code bytes}, they will all be evicted
+   * 
+   * Thread safe.
+   *
+   * @param bytes the size in bytes
+   * @param location the location in block store
+   * @return an eviction plan to achieve the freed space
+   */
+  @Override
+  public EvictionPlan freeSpace(long bytes, BlockStoreLocation location) {
+    List<Pair<Long, BlockStoreLocation>> toMove = new ArrayList<Pair<Long, BlockStoreLocation>>();
+    List<Long> toEvict = new ArrayList<Long>();
+
+    Node p = mHead.nextNode();
+    long evictBytes = 0;
+    // erase race condition with onAccessBlock on internal data structure
+    synchronized (mMeta) {
+      while (p != mTail && evictBytes < bytes) {
+        Node next = p.nextNode();
+        boolean remove = false;
+
+        try {
+          BlockMeta meta = mMeta.getBlockMeta(p.blockId);
+          if (meta.getBlockLocation().belongTo(location)) {
+            evictBytes += meta.getBlockSize();
+            toEvict.add(p.blockId);
+            remove = true;
+          }
+        } catch (IOException ioe) {
+          LOG.warn("Remove block %d from LRU Cache because %s", p.blockId, ioe);
+          remove = true;
+        }
+
+        if (remove) {
+          p.remove();
+          mCache.remove(p.blockId);
+        }
+
+        p = next;
+      }
+    }
+
+    return new EvictionPlan(toMove, toEvict);
+  }
+
+  /**
+   * Thread safe
+   */
+  @Override
+  public void onAccessBlock(long userId, long blockId) {
+    Node node;
+    synchronized (mMeta) {
```

Comment:
I would prefer if it was synchronized on a lock object (ie. `final Object mLock`), since the metadata manager isn't used here.